### DISCUSSION TITLE
New Arska beta version with new UI published for preview and testing

### DISCUSSION THREAD

**Discussion by Olli69** [Netgalleria/arska-node#32] [Announcements]
Hi,
First beta version of the newest Arska release with the new user interface and features is now published. The Documentation is not yet (edit: fully) updated at the Wiki, but user interface is quite self explanatory for testing. Releases notes (there can be caveats) at https://raw.githubusercontent.com/Netgalleria/arska-node/devel-ui/RELEASE-NOTES.md . In addition to the new user interface there are many new variables and rule templates.   Arska supports also energy weather forecast published by FMI (Finland only). 

<img width="929" alt="image" src="https://github.com/Netgalleria/arska-node/assets/1752838/539fb25c-24d2-4a7b-9d20-5c7ddc0e9fad">
  
New Arska version binaries you can find [at the installation page](https://iot.netgalleria.fi/arska-install/) and the new version should be visible also on the automatic installation (could require restart to see it on the list). Code you can find in branch [devel-ui](https://github.com/Netgalleria/arska-node/tree/devel-ui) .  Backup your configuration for later restore!

<img width="637" alt="image" src="https://github.com/Netgalleria/arska-node/assets/1752838/0c886d82-18d3-454b-9c7b-49b8833694ef">

Suomeksikin voi kommentoida... 😃

Olli

EDIT: There are reports that the new version gets stalled a while after automatic update is some conditions. Manual update (firmware.bin and littlefs.bin) has fixed the problem. It is still unclear what causes this.

**Comment by stikstone** [comment]
Moi. Laitan nyt tänne, jos joku tätä lukee.

Mulla meni arska jumiin päivityksen jälkeen. Näytti alkuun toimivan, mutta käytännössä lopetti melko pian päivityksen jälkeen. 
Mitkään muuttuja-arvot eivät päivittyneet, Jopa kellon muuttujat pysyivät paikoillaan.

Päivitin nyt manuaalisesti, ja näyttää taas toimivan. Palailen asiaan, jos takkuaa vielä.

**Reply by Olli69** [reply]
Moi,
Kiitos infosta! Eli automaattipäivityksessä jotain ongelmaa tai sitten joku muu.  Mahdollisesti kaikki tiedostot eivät olleet päivittyneet tai sitten eivät olleet tulleet selaimelle saakka. Pitää seurailla. Automaattipäivitys on kieltämättä ominaisuus, joka on ollut hieman hankala testata kaikissa olosuhteissa nykyisellään ja siihen pitää saada lisää luotettavuutta.

Yleisesti päivityksiin liittyen, ei välttämättä juuri tähän:
- Päivityksen jälkeen voi vielä vielä erikseen tyhjentää [selaimen välimuistin](https://support.streak.com/en/articles/2274524-clear-your-browser-s-cache). Voi myös kokeilla selaimen yksityistilassa (Chromessa incognito), jonka pitäisi ladata kaikki resurssit kerralla.
- Voi tsekata http://ip-soite/status tuleeko sieltä teksti-/json-muotoinen vastaus. Tässä ainakin ts (aikaleimakenttä) pitäisi muuttua joka kerta ja variables "117" sisältämä minuuttiarvon sisältämä kerran minuutissa. Jos nuo näyttäisivät päivittyvän, niin mahdollisesti selaimella pyörivä Javascript-koodi on kaatunut johonkin virheeseen, eikä sivut päivittyneet.

**Reply by stikstone** [reply]
Juu, oli tosiaan jotenkin vempulasti ihan jumissa. Hirveesti en päivityksen jälkeen katsonut asian päälle ja huomasin vasta kun varaajalta ei tullutkaan lämmintä vettä. 
Nyt näyttää taas toimivan ihan normaalisti. 
Tuo lähtöjen näkyminen kuvaajassa on pirun hyvä. Pystyy katsomaan nätisti koska on ollut päällä. 
Alkuun tuntui että pökkii, eikä kaikki toimi, mutta saattoi olla kysymys päivityksessä tulleesta bugista.

**Reply by Corn-101** [reply]
Moro, Aivan mahtavaa, tosi hieno tämä uusi käyttöliittymä! 

Minulla kävi sama juttu, kun stikstonella, että näytti pelaavan ja sitten kaikki jätti päivittymättä. Tein manuaalipäivityksen ja toivottavasti nyt pelaa. Jotain hämyä oli myös "Restore configuration":ssa, kun jätti SMA invertterin asetukset ekalla kertaa palauttamatta. Toki ne sai helposti käsin sinne syötettyä.

Kiva, kun saatiin nuo ch_up* muuttujat. Siitä olisi kysymys. Tuo ch_up_period,min, kuinka pitkä aika on period? Sitten toinen kysymys. Kanavat lienee eroteltu pilkuilla? Muuttujassa 152 ch_up_in_8h,min arvot: 1431,348,1431,0,0,0. Kahdeksassa tunnissa 480minuuttia, joten miten tuota pitäisi oikein tulkita? 150 ch_up_period,min taas [1011,0,1011,0,0,0] ja 153 [1461,376,1460,0,0,0]

**Reply by Olli69** [reply]
Moi,
Epäilen, että tiedostojärjestelmän (littlefs.bin) automaattipäivityksessä voisi olla  jotain ongelmaa, eli vähintään se kannattaa päivittää manuaalisesti, tai molemmatkin. Tuohon hitauteen päivityksen jälkeen olen kanssa joskus törmännyt ja miettinyt syytä.  On tullu mieleen, että korruption sijaan (ainakin joissain tapauksessa), voisi olla että esim. tiedostojärjestelmä tekee jotain järjestelyä alussa ja vetää cpu:n tukkoon. Kiusallista ja pitää selvitellä - positiivista on kyllä, että "päästyään vauhtiin" on toiminut hyvin. 

Asetusten palautuksesssa (restore) aikaisemmasta pitää tosiaan käydä täydentämässä mahdolliset mittari- ja invertteritiedot. Tämä johtuu siitä, että aikaisemmin oli vain yksi mittari, nyt mahdollisuus määritellä verkkomittarointi (Shelly 3EM / HAN P1 Telegram) ja invertteri (SMA, Fronius) eri kohtiin. Myös yhteysasetukset muuttuneet - aikaisemmin oli yksi connect string, nyt IP-osoite, portti, salasana erikseen. Eli tämän takia vaatii tässä versiopäivityksessä asetusten täydentämistä näiltä osin.

Noissa sinun ch_up* arvoissa on kyllä jotain outoa! Eli 8 h:n 480 min tai 24 h:n  1440 ei pitäisi mennä yli (suuruusluokkakin on pyöristysvirhettä isompi). Kanavat on tosiaan erotettu pilkulla ja kun käytetään ehdoissa, niin viitataan suoraan kanavakohtaiseen arvoon - eli et voi ehdossa viitata toisen kanavan ch_up*-arvoon (ei liity virheeseen, mutta muuten infona). Näyttää, että virhettä tulee nimenomaan tuosta muuttujan 150 (eli nykyinen tunti/periodi) älyttömän suuresta arvosta, joka osana summaa vie myös 8h (152) ja 24h (153) arvot  virheellisiksi.

Mun pitää katsoa koodia ja miettiä, missä tilanteessa virheelliset arvot ovat voineet syntyä. Toki jos sinulla mahdollisuus lähettää asetusten backuptiedostot (arska-config...json, josta voit karsia kriittiset salasanat/wifi-tiedot tekstieditorilla) vaikka mailissa olli@netgalleria.fi , niin pääsen nopeammin simuloimaan ainakin sun asetusten osalta. Kyselyllä `http://arskan-ip/status` tuloksessa on channel_history -taulukossa kanavien up-time-arvot minuutteina (kustakin kanavasta tunneittain historiaa alkaen -23h nykyhetkeen).

Noita ch_up*-arvoja ei tietysti ole vielä kovin laajasti testattu  ja hyvä että saadaan virheet pois, sillä nämä muuttujat ovat hyödyllisiä ehdoissa.  Itsellä pyörii kyllä parissakin ympäristössä enkä ole huomannut vastaavaa eli todennäköisesti on joku eroavuus esim asetuksissa.

**Reply by Olli69** [reply]
Terve,
Näyttää tosiaan siltä että 0.99 ja myöhempään versioon päivitettäessä tarvitaan kaapelia, mutta sitten taas seuraavat päivitykset menisivät automaattisemmin. Johtuisi siitä, että nyt on käytössä uudempi kehitysympäristö ja se vaatii ilmeisesti myös kolmannen osion (bootloader) päivityksen, jotta uudempi tiedostotojärjestelmä toimisi oikein. Tätä bootloaderia ei yhtä helposti pysty verkon yli päivittämään ja nykypäivityksessä tätä ei ole valmiin. Tarkennan tätä vielä ja pyrin saamaan päivityksiä selkeämmiksi tältä osin mutta tässä vaiheessa kannattaa varautua kaapelin käyttöön.

Päivitin tuonne nyt beta2-version, jota olen vasta ylimalkaisesti testannut, eli tähän voi tulla vielä fiksauksia. Siinä on karsittu bugeja ja tehostettu etenkin alkuvaiheen toimintaa (paljon vähemmän tiedoston käsittelyä). Yksi isompi muistivuoto myös paikattu. Tukee myös mobiilia portrait modea. @Corn-101 kirjaaman ch_up* -ongelman osalta löysin yhden mahdollisen virhelähteen, joka on korjattu. Mutta en tiedä oliko siinä virheen syy tai mahdollisesti muualla, koska en saanut vielä toistettua virhettä,

**Reply by Corn-101** [reply]
Alpha4 on nyt pyörinyt kutakuinkin siitä asti, kun se julkaistiin ja on toiminut kyllä tosi hyvin. Channel up arvot ovat olleet nyt oikein ja Eleringin hintatiedot tulleet mukisematta. Lisäksi on ollut tosi vakaa, ei ole tainnut buutata kertaakaan itsestään. Vielä, kun tuota ch_up muuttujaa voisi käyttää laskentaan ja sen erotuksella pitää esim. yösähköllä halvimpaan aikaan erotuksen verran kanavaa auki..

**Reply by Olli69** [reply]
Hyvä kuulla! Minullakin on käsitys, että aikaisemmat stabiilisuusongelmat erityisesti käynnistyessä on selvitetty ja testilaitteilla Arska on toiminut nyt vakaasti.

ch_up muuttujia (150, 152, 153) pistäisi kyllä pystyä käyttämään saman kanavan ehtoihin samalla lailla kuin muitakin muuttujia. Äsken vielä tein pientä lisätestiä ja siinä näytti toimivan päälläoloajan rajoitus näillä. Käynnistyksen jälkeen ovat aina nollissa ja toisen kanavan ch_up -arvoihin ei kyllä pysty viittaamaan. Esim [videolla](https://www.youtube.com/watch?v=BFsiXRxTFBo&t=765s) käytin demossa sääntöpohjia 402 ja 403, joissa muuttuja 153 käytössä.

Asennussivulla on nyt käännettynä myös uudempi beta3 ja siinä oli ainakin jotain käyttöliittymään liittyvää viilaus.

**Reply by Corn-101** [reply]
Aivan, sori en ole näköjään ymmärtänyt ihan täysin tuota. Eli tuon videon esimerkissä, jos päivällä kanava onkin ollut päällä vain 200 minuuttia, pitää se yön halvimmalla sähköllä kanavaa päällä vain erotuksen verran eli 40 minuuttia? Olin ymmärryksessä, että jos päivällä ei muuttujan 153 ehto täyty, pitää se yöllä täydet, eli videon tapauksessa neljä tuntia.

**Reply by Olli69** [reply]
Jees, eli pitää tässä tapauksessa yöllä päällä vain jäljellä olevat 40 minuuttia, kunnes 24h laskurin arvo on 240. Laskuri kasvaa siis yöllä 200->240. Käynnistyy ensimmäisenä tuntina, joka on 4 halvimman tunnin joukossa. 

EDIT1:
Tuossa säännössä voi olla kyllä sellainen ongelma, että jos ensimmäisenä yönä ajoittuu halvimmat tunnit aamuyöhön ja seuraavana yön alkuun, niin aikaisemman yön laskurit eivät ole vielä nollaantuneet, kun hinnan puolesta voisi kanavan pitää päällä. Tämä näkyisi niin, että ei ajeta täyttä aikaa jälkimmäisenä yönä (olettaen siis että päivällä ei ole tulleet tunnit täyteen) koska ei ole kulunut 24 h edellisestä ajosta.

EDIT2: 
Testissä uusi muuttuja (157), jossa kanavakohtainen päälläolo nykyiseltä ja kahdelta aikaisemmalta blokilta. Tähän esim. yöllä (block id = 1) edellisen loppuyön lukemat  eivät enää vaikuta kuten 24 h tunnin muuttujan (153) arvoon jossain. Tällä voidaan jatkossa korvata joissain säännöissä muuttuja 153.

**Comment by stikstone** [comment]
moi. Ongelmia tässä entso-e hinnan kanssa, niin mietin, että mitenkäs palaaminen .99 -> .93 beta3 versioon? miten onnistuu, ja pystynkö käyttää .99 backuppia asetusten palautukseen?

**Reply by Olli69** [reply]
Moi,
Johtuu tosiaan siitä, että Entso-E:ltä puuttuu kokonaan sunnuntain 28.5. hinnat ja Arska olettaa että ehjä aikasarja. Eli vanhaan versioon palaaminen ei välttämättä auta (en ole varma kyllä). Sinällään varmistusten pitäisi toimia tuohonkin suuntaan, vaikka en ole pahemmin testannut.

Voit myös kokeilla päivittää uusimpaan alfa3 versioon. Siinä on Suomen (sekä Viro, Latvia,Liettua) hinta-alueella uusi vaihtoehtoinen hintatietojen haku (Elering), joka on kyllä toiminut näinä päivinä ihan ok. Kun se on ollut mulla testikäytössä kaikissa laitteissa, niin en edes huomannut tuota Entso-E:n ongelmaa kuin vasta nyt.

Olli

**Reply by Olli69** [reply]
Moi,
Siinä tuon Entso-E:n puuttuvan datan ongelman lisäksi viimeisimpään kehitysversioon oli näemmä "ilmestynyt" typo koodin siivouksen yhteydessä, jonka takia myös haku tökki. Nyt julkaistu alpha4, jossa pitäisi onnistua myös Entso-E-haku. Nyt tulee kyllä virheilmoituksia ja hintagraafi ei näy (koska Entso-E:n puuttuva data sotkee), mutta periodin muuttujien arvot ja sitä kautta laskenta pitäisi olla kunnossa.

Voit toki myös käyttää Eleringiä.

Olli

**Reply by stikstone** [reply]
jeps. 
Nyt näyttäisi taas toimivan. kokeilin myös Elring haun, ja tuntuisi tosiaan olevan aika paljon kivuttomampi. 
Onko tuo Elring ollut kivuttomampi hintatiedon suhteen? Lähinnä sillä, että kannattaisiko vaihtaa siihen? 

Vaihtohan nyt onnistuu ihan heittämällä, ja Ensto-E api avainkin näyttäisi pysyvän tallessa. Ei tarvitse kaivella mistään jos/kun vaihtaa takaisin.

Hyvä näin.

**Reply by Olli69** [reply]
Moi,
Tuo on tosiaan Viron kantaverkkoyhtiön Eleringin palvelu, joka vaikuttaa kompaktilta ja fiksummalta teknisestikin, mutta ollut vasta tosi lyhyen ajan kokeilussa - taisin tehdä viime viikolla. Mutta jonain päivänä tarjosi ainakin ennen dataa kuin Entso-E, eli mulle on tullut vaikutelma että olisi luotettavampi yleisesti. Eli hyvin lyhyen kokemuksen perusteella suosittelen, mutta älä heitä sitä Entso-E API-avainta vielä saunan pesään.

Olli

**Reply by petpauhau** [reply]
Moi,

Minulla on nyt ollut Arska muutaman päivän tuotantokäytössä. Arska ohjaa Sonoff CH4 relettä, joka on fläshätty Tasmota firmiksellä. Releen osalta sellainen havainto, että vaikka Arskan mielestä releen jokin kanava on off-tilassa, voi rele olla todellisuudessa vetäneenä. Ilmeisesti Arska ei hae/saa feefbackkiä releelltä sen tilasta?

Toinen havainto koskee virheilmoitusta Entsoe hintatietojen hakua "20:55:39 Cannot get price data from Entso-E. Check availability from https://transparency.entsoe.eu/." Tämä virheimoitus on etusivulla jatkuvasti, mutta jostain ne hintatiedot kuitenkin latautuvat päivittäin.

Terveisin
Petri

**Reply by Olli69** [reply]
Moi,
Arska ei tosiaan hae wifi-releen statustietoa vaan oletuksena on että relettä ei ohjaa mikään muu. Jos Arska ei saa yhteyttä releeseen silloin kun pitäisi muuttaa tilaa, niin siitä tulee virheilmoitus, mutta ei uusintayritystä.  Mutta tuo wifi-releiden tila Arskan käynnistyessä voisi kyllä vaatia vielä hiomista. Nyt Arska laittaa GPIO-kanavat alas käynnistyessään, mutta ei wifi-releitä. Katson saisiko wifi-releet toimimaan samalla tavalla.

Virheilmoitusten käsittely on aika yksinkertainen: Viimeisin virheilmoitus näkyy etusivulla ja sen voi kuitata pois näkyvistä oikean yläkulman rastista. Onkohan sivulla siis vanha virheilmoitus vai tuleeko joka päivä uusi. Tämän saat selville kuittaamalla ilmoituksen pois näkyvistä. Kuittaaminen hoituu selaimen evästeillä eli jos kuittaat sen vaikka tietokoneen selaimessa, niin kännykän selaimessa ilmoitus voi vielä näkyä.

Olli

**Reply by petpauhau** [reply]
Kiitos vastauksestasi,

Virheilmoituksia tulee koko ajan uusia ja heti kuittaamisen jälkeen uusi. Olen generoinut kolme uutta API-avainta, mutta ilman parannusta.

Petri

**Reply by Olli69** [reply]
Ok, kiitos tarkennuksesta. Pitää katsoa tuota, jos saan saman toistettua testiympäristössä. Jos hinnat kuitenkin lopulta tulevat, niin API-avaimet ovat varmaan kunnossa.  Oletko muuten kokeilut käyttää tietolähteenä Eleringiä, joka ei vaadi API-avainta? Sinällään Entso-E:kin pitäisi kyllä toimia.

Olli

**Reply by Olli69** [reply]
Moi Petri,
Kävin vielä läpi tuon wifi-releiden alustuksen ja Entso-E-hintahaun ja muutokset on nyt koodissa sekä uusimmassa 0.99.0-alfa7 -versiossa. Nyt siis käsitykseni ja testieni mukaan ongelmat korjattu.

Eli nyt Arskan käynnistyksen yhteydessä, heti kun wifi-yhteys toimii, ajetaan myös wifi-releiden kanavat alas. Eli wifi-releiden ei pitäisi jäädä enää roikkumaan päälle kun Arska käynnistetään uudelleen, kuten on voinut jäädä aikaisemmin. Sinällään Arska ei tutki myöhemmin releiden tiloja, eli oletus on että mikään/kukaan muu ei käy niitä muuttamassa, mutta virheilmoitus näkyy jos wifi-releiden tilan muuttaminen ei onnistu.

Tuo raportoimasi Entso-E-hintahaun ongelma johtui tuoreessa päivityksissä tulleesta kirjoitusvirheestä, joka ei sattuneesta syystä tullut esille testeissä aikaisemmin. Tämän haku tulkittiin virheellisesti epäonnistuneeksi. 

Kiitos @petpauhau paljon kun raportoit🙏! 

Olli

**Reply by petpauhau** [reply]
Kiitos Olli päivityksestä. Raportoimani ongelmat näyttävät poistuneen päivityksen myötä.
Terveisin Petri
